src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -110,32 +107,70 @@ internal async Task SearchAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task SearchCurrentDocumentAsync(CancellationToken cancellationToken)
+        {
+            if (_activeDocument == null)
+                return;
+
+            var project = _activeDocument.Project;
+            var service = _host.GetNavigateToSearchService(project);
+            if (service == null)
+                return;
+
+            await _progress.AddItemsAsync(1, cancellationToken).ConfigureAwait(false);
+            try
+            {
+                await service.SearchDocumentAsync(
+                    _activeDocument, _searchPattern, _kinds,
+                    r => _callback.AddItemAsync(project, r, cancellationToken),
+                    cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                await _progress.ItemCompletedAsync(cancellationToken).ConfigureAwait(false);
+            }
+        }
+
         private async Task SearchAllProjectsAsync(bool isFullyLoaded, CancellationToken cancellationToken)
```

Comment:
the logic below is pretty heavily changed from teh original.  've tried to comment throughout this to make things clear.  it's better to just look at the RHS instead of trying to diff things.